### PR TITLE
feat: [UIE-8054] - DBaaS enhancements 1

### PR DESCRIPTION
**Pull request by corya-akamai**
## Description 📝
Common changes for the V2 enhancements that will be used in create/landing/detail flows

## Changes  🔄
List any change relevant to the reviewer.
- Adding capability to accounts and regions
- Check for either capability to show menu item
- Use V2 to add Beta flag
- Add additional factory data to test flows

## Target release date 🗓️
9/3/24

### Prerequisites
(How to setup test environment)
- Add V2 capability to account factory mocks

### Verification steps
(How to verify changes)
- Beta chip shows for V2 capability

## As an Author I have considered 🤔

*Check all that apply*

- [x] 👀 Doing a self review
- [x] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [x] 🤏 Splitting feature into small PRs
- [x] ➕ Adding a [changeset](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md#writing-a-changeset)
- [x] 🧪 Providing/Improving test coverage
- [x] 🔐 Removing all sensitive information from the code and PR description
- [x] 🚩 Using a feature flag to protect the release
- [ ] 👣 Providing comprehensive reproduction steps
- [ ] 📑 Providing or updating our documentation
- [ ] 🕛 Scheduling a pair reviewing session
- [ ] 📱 Providing mobile support
- [ ] ♿  Providing accessibility support

---
## Commit message and pull request title format standards

> **Note**: Remove this section before opening the pull request
**Make sure your PR title and commit message on squash and merge are as shown below**

`<commit type>: [JIRA-ticket-number] - <description>`

**Commit Types:**
- `feat`: New feature for the user (not a part of the code, or ci, ...).
- `fix`: Bugfix for the user (not a fix to build something, ...).
- `change`: Modifying an existing visual UI instance. Such as a component or a feature.
- `refactor`: Restructuring existing code without changing its external behavior or visual UI. Typically to improve readability, maintainability, and performance.
- `test`: New tests or changes to existing tests. Does not change the production code.
- `upcoming`: A new feature that is in progress, not visible to users yet, and usually behind a feature flag.

**Example:** `feat: [M3-1234] - Allow user to view their login history`

---

### CODE REVIEW

File: packages/manager/src/features/Databases/DatabaseLanding/DatabaseLogo.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+      display="flex"
+      id="aBranding"
+      justifyContent="center"
+      sx={style ? style : { margin: '20px' }}
+    >
+      <Typography sx={{ display: 'inline-block', textAlign: 'center' }}>
+        <BetaChip
+          sx={() => ({
+            backgroundColor: '#727272',
+            color: '#ffffff',
```
Comment (mjac0bs) [inline packages/manager/src/features/Databases/DatabaseLanding/DatabaseLogo.tsx:24]: Do we need hex colors on the BetaChip? We try to use [theme colors](https://linode.github.io/manager/development-guide/02-component-structure.html#component-structure) whenever possible.

File: packages/manager/src/features/Databases/DatabaseLanding/DatabaseLogo.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+interface BrandingProps {
+  style?: React.CSSProperties;
+}
+
+export const DatabaseLogo = ({ style }: BrandingProps) => {
+  return (
+    <Box
+      display="flex"
+      id="aBranding"
```
Comment (mjac0bs) [inline packages/manager/src/features/Databases/DatabaseLanding/DatabaseLogo.tsx:16]:
```suggestion
```
Did we need this?

**Review by mjac0bs — COMMENTED** [review]
Thanks for the contribution @corya-akamai!

I'm in agreement with Jaalah about it being beneficial to be able to control this with a LD [feature flag](https://linode.github.io/manager/development-guide/11-feature-flags.html#feature-flags), as well.

Noting that we've got a unit test failure on `DatabaseResizeCurrentConfiguration.test.tsx` (can replicate this with `yarn test DatabaseResizeCurrentConfiguration`) because of changes to the mocks. It looks like types are impacting the `databaseTypeFactory`'s vcpus in the test.

File: packages/manager/src/features/Databases/DatabaseLanding/DatabaseLogo.tsx (hunk 1 above)
Comment (corya-akamai) [reply packages/manager/src/features/Databases/DatabaseLanding/DatabaseLogo.tsx]: Update to use theme colors, although some of the colors were not listed in the palette.

File: packages/manager/src/features/Databases/DatabaseLanding/DatabaseLogo.tsx (hunk 2 above)
Comment (corya-akamai) [reply packages/manager/src/features/Databases/DatabaseLanding/DatabaseLogo.tsx]: No, removed

**Comment by corya-akamai** [comment]
> Thanks for the contribution @corya-akamai!
> 
> I'm in agreement with Jaalah about it being beneficial to be able to control this with a LD [feature flag](https://linode.github.io/manager/development-guide/11-feature-flags.html#feature-flags), as well.
> 
> Noting that we've got a unit test failure on `DatabaseResizeCurrentConfiguration.test.tsx` (can replicate this with `yarn test DatabaseResizeCurrentConfiguration`) because of changes to the mocks. It looks like types are impacting the `databaseTypeFactory`'s vcpus in the test.

Thanks, I've updated the tests. We have a DBaaS meeting today, I'll ask about adding a feature flag.

**Review by mjac0bs — APPROVED** [review]
I'm approving pending: the couple of fixes to get the unit tests passing and the changeset update. This is looking good otherwise.

File: packages/manager/src/features/Databases/utilities.test.ts [hunk 3]
```diff
@@ -23,7 +23,33 @@ describe('useIsDatabasesEnabled', () => {
...
 
-    await waitFor(() => expect(result.current.isDatabasesEnabled).toBe(true));
+    await waitFor(() => {
+      expect(result.current.isDatabasesEnabled).toBe(true);
+      expect(result.current.isDatabasesV1Enabled).toBe(true);
+      expect(result.current.isDatabasesV2Enabled).toBe(false);
+    });
+  });
+
+  it('should return true for an unrestricted user with the account capability V2', async () => {
```
[the remark this answers is not in the document]
Comment (corya-akamai) [reply packages/manager/src/features/Databases/utilities.test.ts:33]: Updated

File: packages/api-v4/.changeset/pr-10786-added-1723662117687.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+---
+"@linode/api-v4": Added
+---
+
+Add DBaaS V2 capabilites ([#10786](https://github.com/linode/manager/pull/10786))
```
[the remark this answers is not in the document]
Comment (corya-akamai) [reply packages/api-v4/.changeset/pr-10786-added-1723662117687.md:5]: Updated

File: packages/manager/src/features/Databases/utilities.test.ts [hunk 5]
```diff
@@ -23,7 +23,33 @@ describe('useIsDatabasesEnabled', () => {
...
+    );
+
+    const { result } = renderHook(() => useIsDatabasesEnabled(), {
+      wrapper: wrapWithTheme,
+    });
+
+    await waitFor(() => {
+      expect(result.current.isDatabasesEnabled).toBe(true);
+      expect(result.current.isDatabasesV1Enabled).toBe(false);
+      expect(result.current.isDatabasesV2Enabled).toBe(true);
```
[the remark this answers is not in the document]
Comment (corya-akamai) [reply packages/manager/src/features/Databases/utilities.test.ts:57]: Thanks updated!

File: packages/manager/src/features/Databases/DatabaseCreate/DatabaseCreate.tsx [hunk 6]
```diff
@@ -452,7 +452,7 @@ const DatabaseCreate = () => {
             },
           ],
           labelOptions: {
-            suffixComponent: flags.databaseBeta ? (
+            suffixComponent: flags.dbaasV2?.beta ? (
```
[the remark this answers is not in the document]
Comment (corya-akamai) [reply packages/manager/src/features/Databases/DatabaseCreate/DatabaseCreate.tsx:456]: Added a little more margin

File: packages/manager/src/features/Databases/DatabaseLanding/DatabaseLogo.tsx [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+        <BetaChip className={classes.betaChip} component="span" />
+        <span className={classes.logo}>
+          Powered by <Logo />
+        </span>
+      </Typography>
+    </Box>
+  );
+};
+
+export default DatabaseLogo;
```
Comment (bnussman-akamai) [inline packages/manager/src/features/Databases/DatabaseLanding/DatabaseLogo.tsx:45]: Can we remove the default export here? In our codebase, we prefer to use only named exports

File: packages/api-v4/src/databases/types.ts [hunk 8]
```diff
@@ -65,7 +65,7 @@ type MemberType = 'primary' | 'failover';
 
 // DatabaseInstance is the interface for the shape of data returned by the /databases/instances endpoint.
 export interface DatabaseInstance {
-  id: number;
+  id: string;
```
[the remark this answers is not in the document]
Comment (corya-akamai) [reply packages/api-v4/src/databases/types.ts:68]: Reverted back to number. Change is no longer needed.

File: packages/manager/src/features/Databases/DatabaseLanding/DatabaseLogo.tsx (hunk 7 above)
Comment (corya-akamai) [reply packages/manager/src/features/Databases/DatabaseLanding/DatabaseLogo.tsx:45]: Done

**Review by bnussman-akamai — APPROVED** [review]

File: packages/manager/src/features/Support/SupportTickets/SupportTicketProductSelectionFields.tsx
Comment (mjac0bs) [inline packages/manager/src/features/Support/SupportTickets/SupportTicketProductSelectionFields.tsx:1]: @corya-akamai Can we revert these changes too with the [revert](https://github.com/linode/manager/pull/10786/commits/b23641bc1477d009d8a51752be055188a30be5c4#diff-51c333908ec8827cdb95136ad1c4aaad0f67fbee333098e55d016b3ba6fe49b0R68) on the `id` type, for type-safety?

Comment (conalryan) [reply packages/manager/src/features/Support/SupportTickets/SupportTicketProductSelectionFields.tsx:1]: Good catch, reverted.

**Review by jaalah-akamai — APPROVED** [review]

**Comment by mjac0bs** [comment]
@corya-akamai I'll merge this as soon as CI finishes.
